docs/tutorials/Remark.adoc

```diff
@@ -0,0 +1,396 @@
+Each field in a Person is implemented as a separate class (e.g. a `Name` object represents the person's name).
+That means we should add a `Remark` class so that we can use a `Remark` object to represent a remark given to a person.
+
+=== Add a new `Remark` class
+
+Create a new `Remark` in `seedu.address.model.person`. Since a `Remark` is a field that is similar to `Address`, we can reuse a significant bit of code.
+
+A quick copy-paste and search-replace later, you should have something like link:https://github.com/j-lum/addressbook-level35/commit/c8e6048f0d81f2b4b7797e838226da823892129b#diff-af2f075d24dfcd333876f0fbce321f25[this].
+
+=== Make use of `Remark`
+
+Let's change `RemarkCommand` and `RemarkCommandParser` to use the new `Remark` class instead of plain `String`.
+These should be relatively simple changes.
+
+== Add a placeholder element for remark to the UI
+
+Without getting too deep into `fxml`, let's go on a 5 minute adventure to get some placeholder text to show up for each person.
+
+Simply add
+[source, java]
+PersonCard.java
+```
+@FXML
+private Label remark;
+```
+
+to link:https://github.com/j-lum/addressbook-level35/commit/06f9897d7088d22689de478c94612f8bcbff2964#diff-0c6b6abcfac8c205e075294f25e851fe[`seedu.address.ui.PersonCard`].
+`@FXML` is an annotation that marks a private or protected field and makes it accessible to FXML.
+It might sound like Greek to you right now, don't worry -- we will get back to it later.
+
+Then insert
+
+```
+<Label fx:id="remark" styleClass="cell_small_label" text="\$remark" />
+```
+into link:https://github.com/j-lum/addressbook-level35/commit/06f9897d7088d22689de478c94612f8bcbff2964#diff-12580431f55d7880578aa4c16f249e71[`main/resources/view/PersonListCard.fxml`].
+
+That's it! Fire up the application again and you should see something like this:
+
+.$remark shows up in each entry
+image::$Remark.png[]
+
+== Modify `Person` to support a `Remark` field
+
+Since `PersonCard` displays data from a `Person`, we need to update `Person` to get our `Remark` displayed!
+
+=== Modify `Person`
+
+We change the constructor of `Person` to take a `Remark`.
+We will also need to define new fields and accessors accordingly to store our new addition.
+
+=== Update other usages of `Person`
+
+Unfortunately, a change to `Person` will cause other commands to break, you will have to modify these commands to use the updated `Person`!
+
+TIP: Use the `Find Usages` feature in IntelliJ IDEA on the `Person` class to find these commands.
+
+== Updating Storage
+
+AddressBook stores data by serializing `JsonAdaptedPerson` into `json` with the help of an external library -- Jackson.
+Let's update `JsonAdaptedPerson` to work with our new `Person`!
+
+While the changes to code may be minimal, the test data will have to be updated as well.
+
+Check out link:https://github.com/j-lum/addressbook-level35/commit/78b2a7438a9f9d828b7f09e190ff6130790966c0[this commit] to see what the changes entail.
```

Comment:
`PersonBuilder` changes are split across https://github.com/j-lum/addressbook-level3/commit/b03b788d5a462403ea7f4dce9a832b62f729f7c4 and https://github.com/j-lum/addressbook-level3/commit/78b2a7438a9f9d828b7f09e190ff6130790966c0
The commit link under Updating Storage https://github.com/j-lum/addressbook-level3/commit/78b2a7438a9f9d828b7f09e190ff6130790966c0 includes changes for testing

Reply:
Yeah I've changed the commits to reflect that but the links are not updated yet. If there's nothing else wrong with the code we can request for @damithc to pull [this branch](https://github.com/j-lum/addressbook-level3/commits/tutorial-add-remark) in this sem's AB-3 and I'll update all the links to match. Can you help me do a quick once over on the commits on the branch as well?